lib/plugins/aws/provider/awsProvider.js

```diff
@@ -194,6 +194,13 @@ class AwsProvider {
     impl.addEnvironmentProfile(result, `AWS_${stageUpper}`);
 
     result.region = this.getRegion();
+
+    const deploymentBucketObject = this.serverless.service.provider.deploymentBucketObject;
+    if (deploymentBucketObject && deploymentBucketObject.serverSideEncryption
```

Comment:
@Athulus could you shed some light into this?

Why is this check needed?
How would one use the `aws:kms` setting?